server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -248,72 +269,14 @@ public boolean hasField(String path) {
 
     /**
      * Checks whether the document contains a value for the provided path
-     * @param path The path within the document in dot-notation
+     *
+     * @param path           The path within the document in dot-notation
      * @param failOutOfRange Whether to throw an IllegalArgumentException if array is accessed outside of its range
      * @return true if the document contains a value for the field, false otherwise
      * @throws IllegalArgumentException if the path is null, empty or invalid.
      */
     public boolean hasField(String path, boolean failOutOfRange) {
-        FieldPath fieldPath = new FieldPath(path);
```

Comment:
It's great to see how much duplicated code can be removed and how much simpler the IngestDocument code gets by delegating to WriteField that nicely encapsulates field resolution and access.